rotkehlchen/tasks/manager.py

```diff
@@ -536,8 +546,50 @@ def _maybe_update_snapshot_balances(self) -> Optional[list[gevent.Greenlet]]:
         Update the balances of a user if the difference between last time they were updated
         and the current time exceeds the `balance_save_frequency`.
         """
-        with self.database.conn.read_ctx() as cursor:
-            if self.database.should_save_balances(cursor):
+        with self.database.conn.write_ctx() as write_cursor:
+            if self.database.should_save_balances(write_cursor):
+                if CachedSettings().get_settings().auto_detect_tokens is True:
```

Comment:
Can you maybe add a small comment here describing the currrent algorithm for token detection here?